### DISCUSSION TITLE
YAML file for server config

### DISCUSSION THREAD

**Discussion by KritapasSuwannawin** [zilliztech/GPTCache#391] [Q&A] [answered]
I found an example of a yaml file here: https://github.com/zilliztech/GPTCache/tree/main/examples#How-to-use-GPTCache-server. However, it does not give information about what options are available for each config (e.g., what options are available for model_src other than onnx).

Can anyone provide information on all the available options for each config?

Thanks

**Comment by SimFG** [comment, marked as answer]
hi, @KritapasSuwannawin , reference: https://github.com/zilliztech/GPTCache/tree/main/examples#how-to-set-the-embedding-function
for example, if you want to use the `paddlenlp` embedding:
```
model_source:
    paddlenlp
model_config:
    model: ernie-3.0-medium-zh
```
the`model_config` param is the construct params, reference: https://gptcache.readthedocs.io/en/latest/references/embedding.html

**Reply by KritapasSuwannawin** [reply]
Hi,

Thanks for the quick response!

I want to use openai as my model. Here is my current yaml file

```
model_src: openai
model_config:
  api_key: my-api-key
config:
  similarity_threshold: 0.9
```

When I run this command: gptcache_server -s 0.0.0.0 -p 3002 -f gptcache.yml, I got this error: TypeError: Onnx() got an unexpected keyword argument 'api_key'.

What should I put in my yaml file to make this works, and what options are available to configure the "evaluation"?

Thanks

**Reply by SimFG** [reply]
sorry, it's a document typo, `model_src` -> `model_source`
evaluation options: https://github.com/zilliztech/GPTCache/tree/main/examples#how-to-set-the-similarity-evaluation-interface
config: `evaluation_kws`
like:
```
evaluation: onnx
evaluation_kws: 
    model: GPTCache/albert-duplicate-onnx
```
params reference: https://gptcache.readthedocs.io/en/latest/references/similarity_evaluation.html
